test/functional/p2p_dns_seeds.py

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env python3
+# Copyright (c) 2021 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test ThreadDNSAddressSeed logic for querying DNS seeds."""
+
+import itertools
+
+from test_framework.p2p import P2PInterface
+from test_framework.test_framework import BitcoinTestFramework
+
+
+class P2PDNSSeeds(BitcoinTestFramework):
+    def set_test_params(self):
+        self.setup_clean_chain = True
+        self.num_nodes = 1
+        self.extra_args = [["-dnsseed=1"]]
+
+    def run_test(self):
+        self.init_arg_tests()
+        self.existing_outbound_connections_test()
+        self.existing_block_relay_connections_test()
+        self.force_dns_test()
+        self.wait_time_tests()
+
+    def init_arg_tests(self):
+        self.log.info("Check that setting -connect disables -dnsseed by default")
+        self.nodes[0].stop_node()
+        with(self.nodes[0].assert_debug_log(expected_msgs=["DNS seeding disabled"])):
+            self.start_node(0, ["-connect=fakenodeaddr"])
+
+        self.log.info("Check that running -connect and -dnsseed means DNS logic runs.")
+        with(self.nodes[0].assert_debug_log(expected_msgs=["Loading addresses from DNS seed"], timeout=12)):
+            self.restart_node(0, ["-connect=1", "-dnsseed=1"])
+
+        self.log.info("Check that running -forcednsseed and -dnsseed=0 throws an error.")
+        self.nodes[0].stop_node()
+        self.nodes[0].assert_start_raises_init_error(
+            expected_msg="Error: Cannot set -forcednsseed to true when setting -dnsseed to false.",
+            extra_args=["-forcednsseed=1", "-dnsseed=0"],
+        )
+
+        self.log.info("Check that running -forcednsseed and -connect throws an error.")
+        # -connect soft sets -dnsseed to false, so throws the same error
+        self.nodes[0].stop_node()
+        self.nodes[0].assert_start_raises_init_error(
+            expected_msg="Error: Cannot set -forcednsseed to true when setting -dnsseed to false.",
+            extra_args=["-forcednsseed=1", "-connect=fakenodeaddr"],
+        )
+
+        # Restore default bitcoind settings
+        self.restart_node(0)
+
+    def existing_outbound_connections_test(self):
+        # Make sure addrman is populated to enter the conditional where we
+        # delay and potentially skip DNS seeding.
+        self.nodes[0].addpeeraddress("192.0.0.8", 8333)
```

Comment:
hmmm, I tried this out but unfortunately since the `CNetAddr` code specifically recognizes these as invalid addresses ([link](https://github.com/bitcoin/bitcoin/blob/da1c0c64fd094880712d1c4167ad9eb3bb6ffcc6/src/netaddress.cpp#L345-L347)), it prevents being able to test the functionality.

Reply:
Got it! Then `192.0.0.8` (the IPv4 dummy address per [RFC 7600](https://www.rfc-editor.org/rfc/rfc7600.html)) is probably a good alternative to `192.0.2.0/24`.

I checked the [IANA IPv4 Special-Purpose Address Registry](https://www.iana.org/assignments/iana-ipv4-special-registry/iana-ipv4-special-registry.xhtml) and `192.0.0.8` (like `192.0.2.0/24`) is guaranteed to not be in use as valid destination address,.

Reply:
awesome, thanks!